src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTablePModelTest.java

```diff
@@ -0,0 +1,119 @@
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.gui;
+
+import java.io.File;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import com.puppycrawl.tools.checkstyle.gui.MainFrameModel.ParseMode;
+
+public class ParseTreeTablePModelTest {
+
+    private DetailAST tree;
+
+    public static String getPath(String filename) {
+        return "src/test/resources/com/puppycrawl/tools/checkstyle/astprinter/" + filename;
+    }
+
+    public static DetailAST parseFile(File file) throws Exception {
+        final FileContents contents = new FileContents(
+                new FileText(file.getAbsoluteFile(), System.getProperty("file.encoding", "UTF-8")));
+        return TreeWalker.parseWithComments(contents);
+    }
+
+    @Before
+    public void loadTree() throws Exception {
+        tree = parseFile(
+                new File(getPath("InputAstTreeStringPrinterAttributesAndMethodsJavadoc.java")));
+    }
+
+    @Test
+    public void testRoot() {
+        final Object root = new ParseTreeTablePModel(tree).getRoot();
+        final int childCount = new ParseTreeTablePModel(null).getChildCount(root);
+        Assert.assertEquals(childCount, 1);
+    }
+
+    @Test
+    public void testChildCount() {
+        final int childCount = new ParseTreeTablePModel(null).getChildCount(tree);
+        Assert.assertEquals(childCount, 5);
+    }
+
+    @Test
+    public void testChild() {
+        final Object child = new ParseTreeTablePModel(null).getChild(tree, 0);
+        Assert.assertTrue(child instanceof DetailAST);
+        Assert.assertSame(((DetailAST) child).getType(), TokenTypes.MODIFIERS);
```

Comment:
TeamCity(TC) is failing build and point to Object usage.
Please update getChild signature in ParseTreeTablePModel to return certain type.

Please resolve all other TC failures.